File: backend/app/schemas/export.py
"""Pydantic schemas for Export API."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel

from app.models.enums import ExportStatusEnum


class StartExportRequest(BaseModel):
    """POST /exports body."""

    carousel_id: UUID


class StartExportResponse(BaseModel):
    """POST /exports response."""

    export_id: UUID


class ExportResponse(BaseModel):
    """GET /exports/{id} response."""

    id: UUID
    carousel_id: UUID
    status: ExportStatusEnum
    download_url: str | None = None
    error_message: str | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/types/export.ts
/** Status of an export job. */
export type ExportStatus = "pending" | "running" | "done" | "failed";

/** Response of POST /api/v1/exports (202). */
export interface StartExportResponse {
  export_id: string;
}

/** Response of GET /api/v1/exports/{id}. */
export interface ExportResponse {
  id: string;
  carousel_id: string;
  status: ExportStatus;
  download_url?: string;
  error_message?: string;
  created_at: string;
  updated_at: string;
}
